src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs

```diff
@@ -606,6 +711,128 @@ static void Test2(object y, out int x)
         }
 
         [Fact]
+        public void Scope_02()
+        {
+            var text = @"
+public class Cls
+{
+    public static void Main()
+    {
+        Test2(x1, 
+              Test1(out int x1));
+    }
+
+    static object Test1(out int x)
+    {
+        x = 1;
+        return null;
+    }
+
+    static void Test2(object y, object x)
+    {
+    }
+}";
+            var compilation = CreateCompilationWithMscorlib(text, options: TestOptions.ReleaseExe, parseOptions: TestOptions.Regular.WithOutVarFeature());
+
+            compilation.VerifyDiagnostics(
+                // (6,15): error CS0841: Cannot use local variable 'x1' before it is declared
+                //         Test2(x1, 
+                Diagnostic(ErrorCode.ERR_VariableUsedBeforeDeclaration, "x1").WithArguments("x1").WithLocation(6, 15)
+                );
+        }
+
+        [Fact]
+        public void Scope_03()
+        {
+            var text = @"
+public class Cls
+{
+    public static void Main()
+    {
+        Test2(out x1, 
+              Test1(out int x1));
+    }
+
+    static object Test1(out int x)
+    {
+        x = 1;
+        return null;
+    }
+
+    static void Test2(out int y, object x)
+    {
+        y = 1;
+    }
+}";
+            var compilation = CreateCompilationWithMscorlib(text, options: TestOptions.ReleaseExe, parseOptions: TestOptions.Regular.WithOutVarFeature());
+
+            compilation.VerifyDiagnostics(
+                // (6,19): error CS0841: Cannot use local variable 'x1' before it is declared
```

Comment:
Same response.

---
In reply to: [64252509](https://github.com/dotnet/roslyn/pull/11493#discussion_r64252509) [](ancestors = 64252509)